### PR TITLE
add release notes for 7.5.2

### PR DESCRIPTION
**Pull request by pcsanwald**
Release notes for 7.5.2

Preview: http://elasticsearch_51259.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/7.5/release-notes-7.5.2.html

### CODE REVIEW

File: docs/reference/release-notes/7.5.asciidoc [hunk 1]
```diff
@@ -1,16 +1,63 @@
...
+* SQL: change the way unsupported data types fields are handled {pull}50823[#50823]
+* SQL: Optimisation fixes for conjunction merges {pull}50703[#50703] (issue: {issue}49637[#49637])
+* SQL: Fix issue with CAST and NULL checking. {pull}50371[#50371] (issue: {issue}50191[#50191])
+* SQL: fix NPE for JdbcResultSet.getDate(param, Calendar) calls {pull}50184[#50184] (issue: {issue}50174[#50174])
+
+Search::
+* Fix upgrade of custom similarity {pull}50851[#50851] (issue: {issue}50763[#50763])
+
+Security::
+* Always consume the body in has privileges {pull}50298[#50298] (issue: {issue}50288[#50288])
```
Comment (jrodewig) [inline docs/reference/release-notes/7.5.asciidoc]:
```suggestion
* Always consume the request body in `has_privileges` API {pull}50298[#50298] (issue: {issue}50288[#50288])
```
This one if confusing as-is.

File: docs/reference/release-notes/7.5.asciidoc [hunk 2]
```diff
@@ -1,16 +1,63 @@
...
+* Fix meta version of task index mapping {pull}50363[#50363] (issue: {issue}48393[#48393])
+
+Recovery::
+* Check allocation id when failing shard on recovery {pull}50656[#50656] (issue: {issue}50508[#50508])
+
+SQL::
+* SQL: change the way unsupported data types fields are handled {pull}50823[#50823]
+* SQL: Optimisation fixes for conjunction merges {pull}50703[#50703] (issue: {issue}49637[#49637])
+* SQL: Fix issue with CAST and NULL checking. {pull}50371[#50371] (issue: {issue}50191[#50191])
+* SQL: fix NPE for JdbcResultSet.getDate(param, Calendar) calls {pull}50184[#50184] (issue: {issue}50174[#50174])
```
Comment (jrodewig) [inline docs/reference/release-notes/7.5.asciidoc]:
```suggestion
* Fix NPE for JdbcResultSet.getDate(param, Calendar) calls {pull}50184[#50184] (issue: {issue}50174[#50174])
```

File: docs/reference/release-notes/7.5.asciidoc [hunk 3]
```diff
@@ -1,16 +1,63 @@
...
+* Ensure that field collapsing works with field aliases. {pull}50722[#50722] (issues: {issue}32648[#32648], {issue}50121[#50121])
+* Fix meta version of task index mapping {pull}50363[#50363] (issue: {issue}48393[#48393])
+
+Recovery::
+* Check allocation id when failing shard on recovery {pull}50656[#50656] (issue: {issue}50508[#50508])
+
+SQL::
+* SQL: change the way unsupported data types fields are handled {pull}50823[#50823]
+* SQL: Optimisation fixes for conjunction merges {pull}50703[#50703] (issue: {issue}49637[#49637])
+* SQL: Fix issue with CAST and NULL checking. {pull}50371[#50371] (issue: {issue}50191[#50191])
```
Comment (jrodewig) [inline docs/reference/release-notes/7.5.asciidoc]:
```suggestion
*  Fix issue with CAST and NULL checking. {pull}50371[#50371] (issue: {issue}50191[#50191])
```

File: docs/reference/release-notes/7.5.asciidoc [hunk 4]
```diff
@@ -1,16 +1,63 @@
...
+Mapping::
+* Ensure that field collapsing works with field aliases. {pull}50722[#50722] (issues: {issue}32648[#32648], {issue}50121[#50121])
+* Fix meta version of task index mapping {pull}50363[#50363] (issue: {issue}48393[#48393])
+
+Recovery::
+* Check allocation id when failing shard on recovery {pull}50656[#50656] (issue: {issue}50508[#50508])
+
+SQL::
+* SQL: change the way unsupported data types fields are handled {pull}50823[#50823]
+* SQL: Optimisation fixes for conjunction merges {pull}50703[#50703] (issue: {issue}49637[#49637])
```
Comment (jrodewig) [inline docs/reference/release-notes/7.5.asciidoc]:
```suggestion
*  Optimisation fixes for conjunction merges {pull}50703[#50703] (issue: {issue}49637[#49637])
```

File: docs/reference/release-notes/7.5.asciidoc [hunk 5]
```diff
@@ -1,16 +1,63 @@
...
 
+Mapping::
+* Ensure that field collapsing works with field aliases. {pull}50722[#50722] (issues: {issue}32648[#32648], {issue}50121[#50121])
+* Fix meta version of task index mapping {pull}50363[#50363] (issue: {issue}48393[#48393])
+
+Recovery::
+* Check allocation id when failing shard on recovery {pull}50656[#50656] (issue: {issue}50508[#50508])
+
+SQL::
+* SQL: change the way unsupported data types fields are handled {pull}50823[#50823]
```
Comment (jrodewig) [inline docs/reference/release-notes/7.5.asciidoc]:
```suggestion
*Change the way unsupported data types fields are handled {pull}50823[#50823]
```

File: docs/reference/release-notes/7.5.asciidoc [hunk 6]
```diff
@@ -1,16 +1,63 @@
...
 Machine Learning::
 * Fixes potential memory corruption or inconsistent state when background
 persisting categorizer state. {ml-pull}921[#921]
 
+Mapping::
+* Ensure that field collapsing works with field aliases. {pull}50722[#50722] (issues: {issue}32648[#32648], {issue}50121[#50121])
+* Fix meta version of task index mapping {pull}50363[#50363] (issue: {issue}48393[#48393])
+
+Recovery::
+* Check allocation id when failing shard on recovery {pull}50656[#50656] (issue: {issue}50508[#50508])
```
Comment (jrodewig) [inline docs/reference/release-notes/7.5.asciidoc]:
```suggestion
* Check allocation ID when failing shard on recovery {pull}50656[#50656] (issue: {issue}50508[#50508])
```

File: docs/reference/release-notes/7.5.asciidoc [hunk 7]
```diff
@@ -1,16 +1,63 @@
...
+* Account soft-deletes in FrozenEngine {pull}51192[#51192] (issue: {issue}50775[#50775])
+
+Features/ILM+SLM::
+* Fix SLM check for restore in progress {pull}50868[#50868]
+
+Features/Ingest::
+* Fix ingest simulate response document order if processor executes async {pull}50244[#50244]
+
+Features/Java Low Level REST Client::
+* Improve warning value extraction performance in Response {pull}50208[#50208] (issue: {issue}24114[#24114])
```
Comment (jrodewig) [inline docs/reference/release-notes/7.5.asciidoc]:
```suggestion
* Improve warning value extraction performance in response {pull}50208[#50208] (issue: {issue}24114[#24114])
```

File: docs/reference/release-notes/7.5.asciidoc [hunk 8]
```diff
@@ -1,16 +1,63 @@
 [[release-notes-7.5.2]]
 == {es} version 7.5.2
 
-coming::[7.5.2]
+[[enhancement-7.5.2]]
+[float]
+=== Enhancements
+
+Features/Ingest::
+* Foreach processor - fork recursive call  {pull}50514[#50514]
```
Comment (jrodewig) [inline docs/reference/release-notes/7.5.asciidoc]:
```suggestion
* Fork recursive calls in Foreach processor {pull}50514[#50514]
```

**Review by jrodewig — APPROVED** [review]
Left a few nitty comments. Nothing that I'd consider blocking though.

**Review by lcawl — APPROVED** [review]
LGTM2

**Comment by pcsanwald** [comment]
This test failure is a known failure that @lcawl is looking into, so, I'm going to go ahead and merge this PR.

**Pull request merged**